src/tests.c

```diff
@@ -3416,6 +3416,74 @@ void run_group_decompress(void) {
 
 /***** ECMULT TESTS *****/
 
+void test_pre_g_table(const secp256k1_ge_storage * pre_g, size_t n) {
```

Comment:
Took me embarassingly long to understand what is going on here. Perhaps worth adding an explanation like:
```
For all i in [0, n), p = pre_g[i], q = pre_g[i+1], gg = 2*pre_g[0] we have p + gg = q. Instead of more costly group operations, we check that the 3 points p, gg and -q are distinct, on the curve and on the same line.
```

Reply:
I added some more docs.

Reply:
Thanks!